pkg/cassandra/config.go

```diff
@@ -37,129 +36,82 @@ func IsCassandra3(version string) bool {
 }
 
 func isCassandra4(version string) bool {
-	return strings.HasPrefix(version, "4.0")
+	return strings.HasPrefix(version, "4.")
 }
 
 func (c config) MarshalJSON() ([]byte, error) {
 	config := make(map[string]interface{})
 
-	if c.CassandraYaml != nil {
-		if isCassandra4(c.cassandraVersion) {
-			c.StartRpc = nil
-			c.ThriftPreparedStatementCacheSizeMb = nil
-		}
-
-		// Even though we default to Cassandra's stock defaults for num_tokens, we need to
-		// explicitly set it because the config builder defaults to num_tokens: 1
-		if c.NumTokens == nil {
-			if isCassandra4(c.cassandraVersion) {
-				numTokens := 16
-				c.NumTokens = &numTokens
-			} else {
-				numTokens := 256
-				c.NumTokens = &numTokens
-			}
-		}
-
-		config["cassandra-yaml"] = c.CassandraYaml
+	if isCassandra4(c.cassandraVersion) {
+		c.StartRpc = nil
+		c.ThriftPreparedStatementCacheSizeMb = nil
 	}
 
-	if c.JvmOptions != nil {
-		if strings.HasPrefix(c.cassandraVersion, "3.11") {
-			config["jvm-options"] = c.JvmOptions
+	// Even though we default to Cassandra's stock defaults for num_tokens, we need to
+	// explicitly set it because the config builder defaults to num_tokens: 1
+	if c.NumTokens == nil {
+		if isCassandra4(c.cassandraVersion) {
+			numTokens := 16
+			c.NumTokens = &numTokens
```

Comment:
This could be changed to `c.NumTokens = pointer.Int(16)`. I only thought of it since I saw you introduce usage of the `pointer` package elsewhere :)